rust/private/rustc.bzl

```diff
@@ -958,6 +958,8 @@ def construct_arguments(
     compilation_mode = get_compilation_mode_opts(ctx, toolchain)
     rustc_flags.add(compilation_mode.opt_level, format = "--codegen=opt-level=%s")
     rustc_flags.add(compilation_mode.debug_info, format = "--codegen=debuginfo=%s")
+    if toolchain.target_os.startswith("linux") or toolchain.target_os.startswith("none"):
```

Comment:
why is this platform specific, and also that seems like an extremely strange choice of check. why not != macos?

Reply:
If you take a peak at the link I posted in the comments they mention this issue with macos. I can add this as comment if you would like.

I've only tested this with bare-metal and linux and was being defence in to applying to everything besides macos. If bazel only ever has linux, none, and macos I can change the if statement to be more concise.

Reply:
Okay I removed the conditional statement to only strip if its linux or bare metal. Do you have a recommendation for how to approach using its own copy of strip? Its not clear to me how strip is actually being called by rustc.